tests/app/test_jinja_filters.py

```diff
@@ -859,3 +859,39 @@ def test_format_repeated_summaries_delimiters(self):
         self.autoescape_context = Mock(autoescape=True)
         output = format_repeating_summary(self.autoescape_context, [['', '51 Testing Gardens', '', 'Bristol', 'BS9 1AW']], delimiter=', ')
         self.assertEqual(output, '<ul><li>51 Testing Gardens, Bristol, BS9 1AW</li></ul>')
+
+    def test_format_address_list_undefined_values(self):
+        user_entered_address = [Undefined(), Undefined(), Undefined(), Undefined(), Undefined()]
+        metadata_address = ['123', 'Testy', 'Place', 'Newport', 'NP5 7AR']
+        self.assertEqual('123<br />Testy<br />Place<br />Newport<br />NP5 7AR',
+                         format_address_list(user_entered_address, metadata_address))
+
+    def test_format_address_list_missing_values(self):
+        user_entered_address = ['44', 'Testing', '', 'Swansea', '']
+        metadata_address = ['123', 'Testy', 'Place', 'Newport', 'NP5 7AR']
+        self.assertEqual('44<br />Testing<br />Swansea',
+                         format_address_list(user_entered_address, metadata_address))
+
+    def test_format_address_list_None_value(self):
+        user_entered_address = [None, None, None, None, None]
+        metadata_address = [None, None, None, None, None]
+        with self.assertRaises(Exception):
+            format_address_list(user_entered_address, metadata_address)
+
+    def test_format_address_list_no_values_in_answer(self):
+        user_entered_address = ['', '', '', '', '']
```

Comment:
Not sure if this conversation happened, but is it wrong to use metadata in this scenarios?